backup/data.go

```diff
@@ -86,8 +86,20 @@ func CopyTableOut(connectionPool *dbconn.DBConn, table Table, destinationToWrite
 		// process column names to exclude generated columns from data copy out
 		columnNames = ConstructTableAttributesList(table.ColumnDefs)
 	}
+	tableName := table.FQN()
+	ignoreExternalPartitions := " IGNORE EXTERNAL PARTITIONS"
+	if table.ExtensionTableConfig != nil && *table.ExtensionTableConfig != "" {
+		if columnNames == "" {
+			columnNames = ConstructTableAttributesList(table.ColumnDefs)
+		}
+		tableName = fmt.Sprintf("(SELECT %s FROM %s %s)", columnNames[1:len(columnNames)-1], tableName, *table.ExtensionTableConfig)
```

Comment:
what is the expected behaviour for the case:
```
-- inside the test extension script
create schema if not exists test;
create table test.ao1 (a int, b int, c text) 
with (appendonly=true) 
distributed by (a)
partition by range (b) (start (1) end (10) every (1));

select pg_extension_config_dump('test.ao1', '');
select pg_extension_config_dump('test.ao1_1_prt_1', 'WHERE a=1');
```

Reply:
@andr-sokolov said not to consider partitioning in this patch.